puppet-checks/src/main/java/com/iadams/sonarqube/puppet/checks/SingleQuotedStringContainingVariablesCheck.java

```diff
@@ -51,7 +51,7 @@ public void init() {
 
   @Override
   public void visitNode(AstNode node) {
-    if (CheckStringUtils.containsVariable(node.getTokenValue())) {
+    if (CheckStringUtils.containsVariable(node.getTokenValue()) && !("defined".equals(node.getTokenValue()))) {
```

Comment:
Sure, you can jump on https://gitter.im/iwarapter/sonar-puppet if you want to chat.